debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/SnapshotReader.java

```diff
@@ -446,7 +451,50 @@ protected void execute() {
                                         final Object[] row = new Object[numColumns];
                                         while (rs.next()) {
                                             for (int i = 0, j = 1; i != numColumns; ++i, ++j) {
-                                                row[i] = rs.getObject(j);
+                                                Column actualColumn = table.columns().get(i);
+                                                if(actualColumn.jdbcType() == Types.TIME) {
+                                                    // As MySQL connector/J implementation is broken for MySQL type "TIME" we have to use a binary-ish workaround
+                                                    // @see https://issues.jboss.org/browse/DBZ-342
+                                                    Blob b = rs.getBlob(j);
+                                                    String timeString;
+                                                    try {
+                                                        timeString = new String(b.getBytes(1, (int) (b.length())), "UTF-8");
+                                                    } catch (UnsupportedEncodingException e) {
+                                                        logger.error("Could not read MySQL TIME value as UTF-8");
+                                                        throw new RuntimeException(e);
+                                                    }
+                                                    String[] time = timeString.split(":");
```

Comment:
The splitting patterns should be extracted into re-used `Pattern` instances, so to avoid reparsing the regex for each execution.

Reply:
good point. I will do that.

Also, using KC is in comparison more a performance issue... (reading binary data from a binlog or similar, converting them into a Java type, to convert them to a byte array part on Kafka side). As stated above, that would make more sense if you can share the types in a proper implementation for Avro (and only for Avro/Avro Registry). Other serzializers don't need that type binding and deliver their own mapping anyway. Debezium types are used only as a mapping information for KC, especially when Debezium is just a source, then it's the mapping to the type instance on the Kafka side (Also why you don't have this fromLogical/toLogical implementations in the Debezium types, right?). Only Avro needs the original type here for the schema registry stuff. Would be good to decouple that, but this probably goes against the KC behaviour and confluent pushes Avro with that. Guessed.

Reply:
small correction, I checked the implementation of java.lang.String#split(java.lang.String, int) and it uses a much faster algorithm (iterate over the string and match the single char) when the pattern is just a single char. I will leave that implementation then, if you agree? No need for a Pattern.compile in this case, I guess?